components/list/list-item-drag-handle.js

```diff
@@ -0,0 +1,204 @@
+
+import '../button/button-icon.js';
+import '../icons/icon.js';
+import { css, html, LitElement } from 'lit-element/lit-element.js';
+import { DOWN, END, ENTER, ESC, HOME, LEFT, RIGHT, SPACE, TAB, UP } from '../../helpers/keyCodes.js';
+import { buttonStyles } from '../button/button-styles.js';
+import { getFirstFocusableDescendant } from '../../helpers/focus.js';
+
+export const dragActions = Object.freeze({
+	first: 'first',
+	last: 'last',
+	up: 'up',
+	down: 'down',
+	active: 'keyboard-active',
+	save: 'keyboard-deactivate-save',
+	cancel: 'keyboard-deactivate-cancel',
+	nextElement: 'next-element',
+	previousElement: 'previous-element'
+});
+
+class ListItemDragHandle extends LitElement {
+
+	static get properties() {
+		return {
+			disabled: { type: Boolean, reflect: true },
+			text: { type: String },
+			_keyboardActive: { type: Boolean },
+		};
+	}
+
+	static get styles() {
+		return [ buttonStyles, css`
+			:host {
+				align-items: center;
+				display: flex;
+				margin: 0.25rem;
+			}
+			:host([hidden]) {
+				display: none;
+			}
+			.d2l-list-item-drag-handle-dragger-button,
+			.d2l-list-item-drag-handle-keyboard-button {
+				display: grid;
+				grid-auto-rows: 1fr 1fr;
+				margin: 0;
+				min-height: 1.8rem;
+				padding: 0;
+				width: 0.9rem;
+			}
+			/* Firefox includes a hidden border which messes up button dimensions */
+			button::-moz-focus-inner {
+				border: 0;
+			}
+			.d2l-list-item-drag-handle-dragger-button {
+				background-color: unset;
+			}
+			button,
+			button[disabled]:hover,
+			button[disabled]:focus {
+				background-color: var(--d2l-color-gypsum);
+				color: var(--d2l-color-ferrite);
+			}
+			button:hover,
+			button:focus {
+				background-color: var(--d2l-color-mica);
+			}
+			button[disabled] {
+				cursor: default;
+				opacity: 0.5;
+			}
+		`];
+	}
+
+	constructor() {
+		super();
+		this._keyboardActive = false;
+		this.disabled = false;
+	}
+
+	focus() {
+		const node = getFirstFocusableDescendant(this);
+		if (node) node.focus();
+	}
+
+	render() {
+		return html`${this._keyboardActive && !this.disabled ? this._renderKeyboardDragging() : this._renderDragger()}`;
+	}
+
+	updated(changedProperties) {
+		if (changedProperties.has('_keyboardActive')) {
+			this.focus();
+		}
+	}
+
+	_dispatchAction(action) {
+		if (!action) return;
+		this.dispatchEvent(new CustomEvent('d2l-list-item-drag-handle-action', {
+			detail: { action },
+			bubbles: false
+		}));
+	}
+
+	_dispatchActionDown() {
+		this._dispatchAction(dragActions.down);
+	}
+
+	_dispatchActionUp() {
+		this._dispatchAction(dragActions.up);
+	}
+
+	_onBlur() {
+		this._keyboardActive = false;
+		if (!this._tabbing) {
+			this._dispatchAction(dragActions.save);
+		}
+	}
+
+	_handleActiveKeyboard(e) {
+		if (!this._keyboardActive) {
+			return;
+		}
+		let action = null;
+		switch (e.key) {
```

Comment:
I think `keyCode` has been deprecated for a while but we were unable to avoid it, possibly due to IE11.  Mozilla actually says to use `code`, but also mentions that it may not be supported by all browsers.  Is `key`?  Note that we have a whole bunch of other places that use numeric key codes, so in order for them to use the new import, we'll need to make some additional adjustments, assuming all works.

Reply:
Apparently there are differences in key ~ `code` mappings.  Fairly long list of them on Mozilla.

Reply:
So I didn't use code because it apparently maps to buttons not the character. They have a warning for code:

> Warning: This ignores the user's keyboard layout, so that if the user presses the key at the "Y" position in a QWERTY keyboard layout (near the middle of the row above the home row), this will always return "KeyY", even if the user has a QWERTZ keyboard (which would mean the user expects a "Z" and all the other properties would indicate a "Z") or a Dvorak keyboard layout (where the user would expect an "F").
>https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent

This made me want to use key instead of code.

Why did I go against the grain with using keycode? The main reason is that the tests failed because old edge doesn't support building a `KeyboardEvent` with keycode anymore. So all my tests failed in old edge.

I missed this when I first scanned the page.
>     case "Left": // IE/Edge specific value

However, I don't have this and `case "arrowLeft";` seemed to have worked. I am going to go back a double check that this actually works in the browser. Worried I got excited to finally fixed the tests and didn't check.

If this doesn't work, I guess using `code` would be best? I really don't like that warning. 

The other option is to code up a helper method that does this for us and we never have to worry about it again. My issue with that right now is the only way I can see that working well is having a messy signature for the method.

Reply:
So I think this is growing bigger then this PR and should be thought more about before any patterns are made in core about this. I found that Margaree has fixed my edge problems with tests. So I am going to copy what Calendar did.